lib/api/include/irods/ticket_administration.hpp

```diff
@@ -0,0 +1,598 @@
+#ifndef IRODS_TICKET_ADMINISTRATION_HPP
+#define IRODS_TICKET_ADMINISTRATION_HPP
+
+#undef RxComm
+
+#ifdef IRODS_TICKET_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+#    define RxComm RsComm
+
+struct RsComm;
+#else
+#    define RxComm RcComm
+
+struct RcComm;
+#endif // IRODS_TICKET_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+
+#include <string_view>
+#include <type_traits>
+#include <string>
+
+///
+/// \brief Namespace to hold the ticket administration features
+///
+namespace irods::experimental::administration::ticket
+{
+    ///
+    /// \brief Struct tag to indicate admin privilige
+    ///
+    inline struct admin_tag
+    {
+    } admin;
+
+    ///
+    /// \brief Enumeration to indicate the ticket type
+    ///
+    enum class type
```

Comment:
Bringing attention to the fact that this changed since @korydraughn had opinions

Reply:
Will be changing back to the original version (ticket_type)